lib/active_model/serializer/reflection.rb

```diff
@@ -75,10 +75,10 @@ def value(serializer)
 
         if block
           block_value = instance_exec(serializer, &block)
-          if block_value == :nil
-            serializer.read_attribute_for_serialization(name)
-          else
+          if block_value != :nil
             block_value
+          elsif @_include_data
```

Comment:
@NullVoxPopuli please don't merge code with missing `else` clauses.  I was going to say it was a bug and raise a ❗ but then I realized that it was the feature of the PR. 😱   :(